janusgraph-core/src/main/java/org/janusgraph/util/datastructures/SetUtil.java

```diff
@@ -0,0 +1,51 @@
+// Copyright 2020 JanusGraph Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package org.janusgraph.util.datastructures;
+
+import java.util.HashSet;
+import java.util.Set;
+
+/**
+ * @author Florian Grieskamp (Florian.Grieskamp@gdata.de)
+ */
+public class SetUtil {
+    
+    public static <T> Set<Set<T>> powerSet(final Set<T> originalSet) {
```

Comment:
Can we maybe follow guava's powerSet contract? They look reasonable to me and nice to have IMHO.
```
IllegalArgumentException: If set has more than 30 unique elements, because this causes the power set size to exceed the int range.
NullPointerException: If set is or contains null.
```